src/go/rpk/pkg/cli/redpanda/admin/brokers/list.go

```diff
@@ -26,27 +50,88 @@ func newListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 			bs, err := cl.Brokers(cmd.Context())
 			out.MaybeDie(err, "unable to request brokers: %v", err)
 
-			headers := []string{"Node-ID", "Num-Cores", "Membership-Status"}
+			headers := []string{"ID", "Host", "Port", "Rack", "Cores", "Membership", "Is-Alive", "Version"}
 
 			args := func(b *rpadmin.Broker) []interface{} {
-				ret := []interface{}{b.NodeID, b.NumCores, b.MembershipStatus}
+				ret := []interface{}{b.NodeID, b.InternalRPCAddress, b.InternalRPCPort, checkEmpty(b.Rack), b.NumCores, b.MembershipStatus, *b.IsAlive, extractVersion(b.Version)}
 				return ret
 			}
-			for _, b := range bs {
-				if b.IsAlive != nil {
-					headers = append(headers, "Is-Alive", "Broker-Version")
-					orig := args
-					args = func(b *rpadmin.Broker) []interface{} {
-						return append(orig(b), *b.IsAlive, b.Version)
-					}
-					break
+
+			idUUIDMapping, err := cl.GetBrokerUuids(cmd.Context())
+			if err != nil {
+				fmt.Printf("unable to retrieve node UUIDs: %v", err)
+			}
+			if idUUIDMapping != nil {
+				headers = append(headers, "UUID")
+				org := args
+				args = func(b *rpadmin.Broker) []interface{} {
+					return append(org(b), mapUUID(b.NodeID, idUUIDMapping))
 				}
 			}
+
 			tw := out.NewTable(headers...)
 			defer tw.Flush()
 			for _, b := range bs {
 				tw.Print(args(&b)...)
 			}
+
+			if all && idUUIDMapping != nil {
+				decomNodes := extractDecomNodes(bs, idUUIDMapping)
+				for _, b := range decomNodes {
+					tw.Print(b.NodeID, "-", "-", "-", "-", "-", "-", "-", b.UUID)
+				}
+			}
 		},
 	}
+	cmd.Flags().BoolVarP(&all, "all", "a", false, "If true, include decommissioned brokers")
+	return cmd
+}
+
+// mapUUID returns a UUID from "mapping" which node ID maps to "nodeID".
+func mapUUID(nodeID int, mapping []rpadmin.BrokerUuids) string {
+	var UUIDs []string
+	for _, node := range mapping {
+		if nodeID == node.NodeID {
+			UUIDs = append(UUIDs, node.UUID)
+		}
+	}
+	if len(UUIDs) == 0 {
+		return "UUID Not Found"
+	}
+	return strings.Join(UUIDs, ", ")
+}
+
+// extractVersion gets rid of a commit hash from a version string.
+// e.g. Given "v24.2.3 - be49068f63acc31d34b9d186cdfa62087fd99222", "v24.2.3" returns.
+func extractVersion(fullVersion string) string {
+	re := regexp.MustCompile(`v\d+\.\d+\.\d+`)
+	return re.FindString(fullVersion)
+}
```

Comment:
Can you try using the `version` package? https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/redpanda/version.go